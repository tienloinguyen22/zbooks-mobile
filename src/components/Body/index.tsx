import { Body } from 'native-base';

export { Body };
