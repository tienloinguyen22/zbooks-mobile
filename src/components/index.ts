export * from './Body';
export * from './Button';
export * from './Card';
export * from './CardItem';
export * from './Container';
export * from './Content';
export * from './DatePicker';
export * from './ErrorBoundary';
export * from './ErrorText';
export * from './Field';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Image';
export * from './Input';
export * from './Item';
export * from './Label';
export * from './Left';
export * from './List';
export * from './ListItem';
export * from './Loading';
export * from './Picker';
export * from './Right';
export * from './ScrollView';
export * from './Text';
export * from './Touchable';
export * from './Thumbnail';
export * from './Tooltip';
export * from './View';
export * from './WithStore';
export * from './WithScreen';
export * from './Alert';
export * from './Animation';
export * from './AlertContainer';
