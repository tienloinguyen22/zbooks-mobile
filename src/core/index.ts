export * from './i18n';
export * from './themes/colors';
export * from './interfaces/ScreenProps';
export * from './interfaces/LoginType';
export * from './interfaces/User';
export * from './interfaces/Resource';
export * from './interfaces/Theme';
export * from './helpers/show_notification';
export * from './helpers/sleep';
export * from './helpers/record_error';
export * from './helpers/configure_google_sign_in';
export * from './helpers/combine_styles';
export * from './helpers/configure_google_sign_in';
export * from './helpers/handle_error';
export * from './helpers/get_primary_color';
export * from './helpers/format_date';
export * from './screen_names';
