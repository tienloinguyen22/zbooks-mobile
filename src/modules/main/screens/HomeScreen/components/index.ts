export * from './NavigationSample';
export * from './IconSample';
export * from './RematchSample';
export * from './NotificationSample';
export * from './CrashSample';
export * from './PickerSample';
export * from './AnalyticsSample';
export * from './TextStyleSample';
