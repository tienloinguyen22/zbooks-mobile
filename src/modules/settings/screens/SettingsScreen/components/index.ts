export * from './SettingMenuItem';
export * from './LogoutModal';
