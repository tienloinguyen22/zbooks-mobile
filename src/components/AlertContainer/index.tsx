import React from 'react';
import { ViewStyle } from 'react-native';
import Modal from 'react-native-modal';
import { combineStyles, colors, THEME_DARK } from '@app/core';
import { useTranslation } from 'react-i18next';
import { useTheme } from '@app/hooks';
import { Icon } from '../Icon';
import { Button } from '../Button';
import { View } from '../View';
import { styles } from './styles';
import { Text } from '../Text';

export type AlertType = 'SUCCESS' | 'WARNING' | 'ERROR' | 'INFO' | 'DEFAULT';

export interface AlertAction {
  title: string;
  onPress: () => void;
  outline?: boolean;
}

interface AlertProps {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose: () => void;
  closeable?: boolean;
}

export const AlertContainer = (props: AlertProps): JSX.Element => {
  const { primaryColor, theme } = useTheme();
  const { t } = useTranslation();

  const backgroundAlertStyle = {
    backgroundColor: theme === THEME_DARK ? colors.lightBlack : colors.white,
  };

  const messageTextStyle = {
    color: theme === THEME_DARK ? colors.white : colors.black,
  };

  const closeModal = (): void => {
    props.closeable !== false && props.onPressClose();
  };

  let titleContainerStyle = combineStyles<ViewStyle>(styles.titleDefault, {
    backgroundColor: primaryColor,
  });
  let title = t('dialog.info');
  let iconName = 'information-outline';

  switch (props.type) {
    case 'WARNING':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.orange,
      });
      title = t('dialog.warning');
      iconName = 'alert-circle-outline';
      break;
    case 'ERROR':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.red,
      });
      title = t('dialog.error');
      iconName = 'alert-outline';
      break;
    case 'SUCCESS':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.green,
      });
      title = t('dialog.success');
      iconName = 'check';
      break;
    default:
      break;
  }

  props.title && ({ title } = props);

  return (
    <View style={styles.container}>
      <Modal isVisible={true} onBackButtonPress={closeModal} onBackdropPress={closeModal}>
        <View style={[styles.alertContainer, backgroundAlertStyle]}>
          <View style={titleContainerStyle}>
            <Icon name={iconName} color={colors.white} size={25} />
            <Text style={styles.titleText}>{title}</Text>
            <View style={styles.dialogIcon} />
          </View>
          <Text style={[styles.messageText, messageTextStyle]}>{props.message}</Text>
          {props.actions && (
            <View style={styles.buttonContainer}>
              {props.actions.map((item) => {
                return (
                  <Button outline={item.outline} onPress={item.onPress} key={item.title} style={styles.buttonDialog}>
                    <Text white={!item.outline} primary={item.outline}>
                      {item.title}
                    </Text>
                  </Button>
                );
              })}
            </View>
          )}
          {!props.actions && (
            <View style={styles.reserveContainer}>
              <Button onPress={props.onPressClose} style={styles.buttonDialog}>
                <Text white>{t('dialog.close')}</Text>
              </Button>
            </View>
          )}
        </View>
      </Modal>
    </View>
  );
};
