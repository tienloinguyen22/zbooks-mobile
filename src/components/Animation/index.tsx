import React from 'react';
import { Resource } from '@app/core';
import LottieView from 'lottie-react-native';

interface Props {
  width: number;
  height: number;
  source: Resource;
  autoPlay?: boolean;
  loop?: boolean;
}

export const Animation = (props: Props): JSX.Element => {
  return (
    <LottieView
      source={props.source}
      style={{
        width: props.width,
        height: props.height,
      }}
      autoPlay={props.autoPlay === true}
      loop={props.loop === true}
    />
  );
};
