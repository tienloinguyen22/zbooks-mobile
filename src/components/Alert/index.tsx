import React from 'react';
import RootSiblings from 'react-native-root-siblings';
import { AlertContainer, AlertAction, AlertType } from '../AlertContainer';

interface ShowParams {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose?: () => void;
  closeable?: boolean;
}

let alert: RootSiblings | undefined;
const hide = (): void => {
  alert && alert.destroy();
};

const show = (props: ShowParams): RootSiblings => {
  alert && hide();
  alert = new RootSiblings(<AlertContainer {...props} onPressClose={props.onPressClose || hide} />);
  return alert;
};

export const Alert = {
  show,
  hide,
};
