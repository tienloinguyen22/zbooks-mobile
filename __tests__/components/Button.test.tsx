import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Button, Text } from '@app/components';
import { THEME_DARK, THEME_LIGHT } from '@app/core';
import { mockTheme } from '../test_helpers';

describe('components/Button', () => {
  const props = {
    onPress: () => {},
  };
  const buttonText = 'Button';

  beforeEach(() => {
    mockTheme(THEME_DARK);
  });

  it('renders successfully', async () => {
    const { baseElement } = render(
      <Button {...props}>
        <Text>{buttonText}</Text>
      </Button>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('renders successfully with light theme', async () => {
    mockTheme(THEME_LIGHT);
    const { baseElement } = render(
      <Button {...props}>
        <Text>{buttonText}</Text>
      </Button>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('renders outline button successfully', async () => {
    const { baseElement } = render(<Button {...props} outline />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders disabled button successfully', async () => {
    const { baseElement } = render(<Button {...props} disabled />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders rounded button successfully', async () => {
    const { baseElement } = render(<Button {...props} rounded />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders success button successfully', async () => {
    const { baseElement } = render(<Button {...props} success />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders info button successfully', async () => {
    const { baseElement } = render(<Button {...props} info />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders warning button successfully', async () => {
    const { baseElement } = render(<Button {...props} warning />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders danger button successfully', async () => {
    const { baseElement } = render(<Button {...props} danger />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders transparent button successfully', async () => {
    const { baseElement } = render(<Button {...props} transparent />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders undefined children successfully', async () => {
    const { baseElement } = render(
      <Button {...props} width={100}>
        {undefined}
        <Text>{buttonText}</Text>
      </Button>,
    );

    expect(baseElement).toMatchSnapshot();
  });
  it('renders with specified width successfully', async () => {
    const { baseElement } = render(
      <Button {...props} width={100}>
        <Text>{buttonText}</Text>
      </Button>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('calls function when click button', async () => {
    const onPress = jest.fn();
    const { getByText } = render(
      <Button {...props} onPress={onPress}>
        <Text>{buttonText}</Text>
      </Button>,
    );
    const button = getByText(buttonText);

    fireEvent.press(button);

    expect(onPress).toBeCalledTimes(1);
  });
});
