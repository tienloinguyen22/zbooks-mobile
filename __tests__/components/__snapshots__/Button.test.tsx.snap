// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button renders danger button successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        danger={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FC4173",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders disabled button successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        disabled={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#8B97AC",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders info button successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        info={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1A97FC",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders outline button successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        outline={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "black",
            "borderRadius": 20,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders rounded button successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        rounded={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders success button successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#24DF98",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
        success={true}
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textTransform": "none",
            }
          }
        >
          Button
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders successfully with light theme 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textTransform": "none",
            }
          }
        >
          Button
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders transparent button successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
        transparent={true}
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders undefined children successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 100,
          }
        }
        width={100}
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textTransform": "none",
            }
          }
        >
          Button
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders warning button successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FDA929",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 200,
          }
        }
        warning={true}
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`components/Button renders with specified width successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "width": 100,
          }
        }
        width={100}
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "textTransform": "none",
            }
          }
        >
          Button
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
